### PR TITLE
Start work on the standard.

### PR DESCRIPTION
**Pull request by DavidSagan**
A quick start on the Standard.

### CODE REVIEW

**Comment by DavidSagan** [comment]
The idea behind element parameter groups is that a given group can be used with multiple element types. This has several advantages. For one, It enforces a naming conformity between element types. Another advantage is that it organizes element parameters of like kind which simplifies documentation.

File: standard/element_parameter.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Element Parameter Naming
+
+Lattice elements parameters are organized into **parameter groups**. All groups are organized as abstract syntax trees.
+At the top level, there are the groups with names like `MagMultipoleGroup`, `ElecMultipoleGroup`, `RFGroup, `BendGroup`, `AlignmentGroup`, etc. 
+Group names use Upper camel case and have `Group` at the end of the name.
+
+A dot `.` is used as a separator between levels in a group. 
+For example, `BendGroup.angle`, `BendGroup.e1`, `RFGroup.frequency`, etc. Using lower case for second and lower level names is encouraged but not mandated.
+
+There are two element parameters that are so common that they are not grouped. These element parameters are `L` (element length) and `Name` (name of element).
```
Comment (youssefhayek) [inline standard/element_parameter.md:10]: Apertures are also common to every element in the lattice.
At the top, there should be a list of constants, such as the length of the lattice (ring =circumference, radius). For the group elements, as well as the length and aperture, maybe the strengts.

File: standard/standard.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+The basic building block used to describe an accelerator is the lattice \vn{element}. Typically,
+a lattice element is something physical like a bending magnet or an electrostatic
+quadrupole, or a diffracting crystal. A lattice element may define a region in space 
+distinguished by the presense of (possibly time-varying) electromagnetic fields,
+materials, apertures and other possible engineered structures. However, lattice elements
+are not restricted to being something physical and may, for example, just mark a particular point in space
+(EG: **Marker** elements), or may designate where beam lines intersect (**Fork** elements).
+By convention, element names in ALS will be upper camel case.
+
```
Comment (youssefhayek) [inline standard/standard.md:57]: The order of the elements must correspond to the physical sequence.

File: standard/standard.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+Essentially, a **lattice**, has an array of **branches** with each branch describing part of the
+machine. Branches can be interconnected to form a unified whole.
+Branches can be interconnected using **Fork** elements. 
+This is used to simulate forking beam lines such as a connections to a transfer line, dump line, or an
+X-ray beam line. The **branch** from which other **branches** fork but is not forked to by any
+other branch is called a **root** branch.
+
+A lattice may contain multiple **root branches. For example, a pair of intersecting storage
+rings will generally have two root branches, one for each ring.
```
Comment (youssefhayek) [inline standard/standard.md:80]: I would suggest creating a definition block for all elements that appear more than twice in the lattice, and a lattice structure for the ordered elements.

File: standard/element_parameter.md (hunk 1 above)
Comment (DavidSagan) [reply standard/element_parameter.md:10]: Apertures are common to most elements but there can be elements that don't have apertures. Fiducial markers for example. 

The PR here is just a start on the section of the standard that describes element attributes. Stuff like specifying the lattice length would go elsewhere. So I do not see anything to change here. what are your thoughts?

File: standard/standard.md (hunk 2 above)
Comment (DavidSagan) [reply standard/standard.md:57]: I put in some words on this.

File: standard/standard.md (hunk 3 above)
Comment (DavidSagan) [reply standard/standard.md:81]: This will need development in future PRs

**Comment by DavidSagan** [comment]
@youssefhayek Please look at the responses and the text change and continue with your review.

File: standard/standard.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+## What ALS Is
+
+ALS is a schema that defines things like the names of various lattice element types, how to organize lattice
+elements into lines which beams of particles or X-rays can move through, etc. 
+
+## What ALS Is Not
+
+ALS does not define any particular grammer to implemement the ALS schema. Rather, there are associated
+language specific standards that define grammers for YAML, JSON, Python, etc. Along with these
```
Comment (cemitch99) [inline standard/standard.md:32]:
```suggestion
language specific standards that define grammars for YAML, JSON, Python, etc. Along with these
```

File: standard/standard.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119).
+
+All `keywords` in this standard are case-sensitive.
+
+## Lattice Elements
+
+The basic building block used to describe an accelerator is the lattice \vn{element}. Typically,
+a lattice element is something physical like a bending magnet or an electrostatic
+quadrupole, or a diffracting crystal. A lattice element may define a region in space 
+distinguished by the presense of (possibly time-varying) electromagnetic fields,
```
Comment (cemitch99) [inline standard/standard.md:52]:
```suggestion
distinguished by the presence of (possibly time-varying) electromagnetic fields,
```

**Review by youssefhayek — APPROVED** [review]

File: standard/standard.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+(EG: **Marker** elements), or may designate where beam lines intersect (**Fork** elements).
+By convention, element names in ALS will be upper camel case.
+
+
+## Lattice Branches
+
+A lattice **Branch** is essentially an ordered array of lattice elements that gives the physical
+sequence to be tracked through.
+In the simplist case, a program can track through the elements one element at a time.
+However, lattice elements may overlap which will naturally complicat tracking.
```
[the remark this answers is not in the document]
Comment (DavidSagan) [reply standard/standard.md:64]: Accepted change.

File: standard/standard.md (hunk 5 above)
Comment (DavidSagan) [reply standard/standard.md]: Accepted change.

**Comment by DavidSagan** [comment]
@cemitch99 Are there any outstanding issues here?

**Review by cemitch99 — APPROVED** [review]

**Pull request merged**